src/external/signal.m4

```diff
@@ -1 +1,4 @@
 AC_CHECK_FUNCS(sigprocmask sigblock sigaction getpgrp prctl)
+
+# Check for the procctl facility found on FreeBSD
+AC_CHECK_HEADERS([sys/procctl.h])
```

Comment:
It feels a bit inconsistent that in one case we check for a function name, in another case - for a header.
But might be ok.

Reply:
I'd also prefer to have a unified approach.

Reply:
How should I resolve this?

Reply:
If you want to resolve this fast, then let's keep `AC_CHECK_FUNCS` for everything.

I'm not really well versed in m4, and my rationale is: if it's good enough for the current code then it's good enough to expand.

Reply:
> > I feel it is the `prctl()` check that should be turned into a header check
> 
> Why?

A function may be present on different OSes, but may be declared in different headers. So simply knowing that a function is available to use doesn't really give that much information to the code that is going to call it.